docs/src/literate/quasi_incompressible_hyperelasticity.jl

```diff
@@ -0,0 +1,362 @@
+        global_dofsu = global_dofs[1:nu]; # first nu dofs are displacement
+        global_dofsp = global_dofs[nu + 1:end]; # last np dofs are pressure
+        @assert size(global_dofs, 1) == nu + np # sanity check
+        ue = w[global_dofsu] # displacement dofs for the current cell
+        pe = w[global_dofsp] # pressure dofs for the current cell
+        @timeit "element assemble" assemble_element!(ke, fe, cell, cellvalues_u, cellvalues_p, mp, ue, pe)
+        assemble!(assembler, global_dofs, fe, ke)
+    end
+end;
+
+# We now define a main function `solve`. For nonlinear quasistatic problems we often like to parameterize the 
+# solution in terms of a pseudo time like parameter, which in this case is used to gradually apply the boundary
+# displacement on the right face. Also for definitenessm we consider λ/μ = 10⁴
+function solve(interpolation_u, interpolation_p)
+    reset_timer!()
+
+    ## import the mesh
+    grid = importTestGrid()
+
+    ## Material parameters
+    μ = 1.
+    λ = 1.E4 * μ
+    mp = NeoHooke(μ, λ)
+
+    ## Create the DofHandler and CellValues
+    dh = create_dofhandler(grid, interpolation_u, interpolation_p)
+    cellvalues_u, cellvalues_p, facevalues_u = create_values(interpolation_u, interpolation_p)
+
+    ## Create the DirichletBCs
+    dbc = create_bc(dh)
+
+    ## Pre-allocation of vectors for the solution and Newton increments
+    _ndofs = ndofs(dh)
+    w  = zeros(_ndofs)
+    ΔΔw = zeros(_ndofs)
+    apply!(w, dbc)
+
+    ## Create the sparse matrix and residual vector
+    K = create_sparsity_pattern(dh)
+    f = zeros(_ndofs)
+
+    ## We run the simulation parameterized by a time like parameter. `Tf` denotes the final value
+    ## of this parameter, and Δt denotes its increment in each step
+    Tf = 2.0;
+    Δt = 0.1;
+    NEWTON_TOL = 1e-8
+
+    pvd = paraview_collection("hyperelasticity_incomp_mixed.pvd");
+    for t ∈ 0.0:Δt:Tf
+        ## Perform Newton iterations
+        print("Time is $t")
+        Ferrite.update!(dbc, t)
+        apply!(w, dbc)
+        newton_itr = -1
+        prog = ProgressMeter.ProgressThresh(NEWTON_TOL, "Solving:")
+        fill!(ΔΔw, 0.0);
+        while true; newton_itr += 1
+            assemble_global!(K, f, cellvalues_u, cellvalues_p, dh, mp, w)
+            norm_res = norm(f[Ferrite.free_dofs(dbc)])
+            apply_zero!(K, f, dbc)
+            ProgressMeter.update!(prog, norm_res; showvalues = [(:iter, newton_itr)])
+
+            if norm_res < NEWTON_TOL
+                break
+            elseif newton_itr > 30
+                error("Reached maximum Newton iterations, aborting")
+            end
+            ## Compute the incremental `dof`-vector (both displacement and pressure)
+            ΔΔw .= K\f;
+
+            apply_zero!(ΔΔw, dbc)
+            w .-= ΔΔw
+        end;
+
+        ## Save the solution fields
+        @timeit "export" begin
+            vtk_grid("hyperelasticity_incomp_mixed_$t.vtu", dh) do vtkfile
```

Comment:
This function has a memory leak (see https://github.com/jipolanco/WriteVTK.jl/issues/43) and can blow up memory usage by accident. Could you add the argument `compress=false`, which is a hotfix around this issue.

Reply:
Just so that I understand it correctly, you mean on each [call to `vtk_grid`](https://github.com/jipolanco/WriteVTK.jl/issues/43#issuecomment-418080118) right?

Reply:
Yes, correct. :+1:

Reply:
Done